src/core/deviceio/cpp/schema_tracker.cpp

```
@@ -58,24 +58,37 @@ std::vector<std::string> SchemaTracker::get_required_extensions()
+        return false;
+    }
+    if (new_index != m_target_collection_index)
+    {
+        m_target_collection_index = new_index;
+        m_last_sample_index.reset();
         std::cout << "Found target collection at index " << *m_target_collection_index << std::endl;
     }
+    return true;
+}
+
+bool SchemaTracker::read_all_samples(std::vector<SampleResult>& samples)
```

Cppcheck (2.19.0)

[style] The function 'read_all_samples' is never used.

(unusedFunction)